src/hotspot/share/oops/klass.hpp

```diff
@@ -580,6 +580,7 @@ class Klass : public Metadata {
   virtual bool should_be_initialized() const    { return false; }
   // initializes the klass
   virtual void initialize(TRAPS);
+  virtual void initialize_preemptable(TRAPS);
```

Comment:
Can we not define these on `instanceKlass` instead of `klass`? Seems really odd to declare virtual methods for which the base class version should never be called (they should be pure virtuals in that case I would think?).

Reply:
Would an initialize_preemptable() = 0 be better?  then you can see if it's called by anything other than InstanceKlass.  Klass is always abstract.